w3-filecoin.md

```diff
@@ -0,0 +1,658 @@
+
+```json
+{
+  "iss": "did:web:web3.storage",
+  "aud": "did:key:agg...",
+  "att": [{
+    "with": "did:web:web3.storage",
+    "can": "piece/add",
+    "nb": {
+      "piece": { "/": "commitment...car" }, /* commitment proof for piece */
+      "storefront": "did:web:web3.storage", /* storefront responsible for invocation */
+      "group": "did:web:free.web3.storage", /* grouping of joining segments into an aggregate */
+    }
+  }],
+  "prf": [],
+  "sig": "..."
+}
+```
+
+Aggregator MUST issue a signed receipt to acknowledge the received request. Issued receipt MUST contain an [effect](https://github.com/ucan-wg/invocation/#7-effect) with a subsequent task (`.fx.join` field) that is run when piece is added to an aggregate and either succeeds (implying that aggregate was queued for being offered) or fails (with `error` describing the problem).
+
+```json
+{
+  "ran": "bafy...invocation",
+  "out": {
+    "ok": {
+      "status": "queued"
+    }
+  },
+  "fx": {
+    "join": { "/": "bafy...dequeue" }
+  },
+  "meta": {},
+  "iss": "did:key:agg...",
+  "prf": []
+}
+```
+
+When piece request to be added is dequeued, aggregator should invoke `piece/add` to include it in an aggregate.
+
+> `did:key:agg...` invokes capability from `did:key:agg...`
+
+```json
+{
+  "iss": "did:key:agg...",
+  "aud": "did:key:agg...",
+  "att": [{
+    "with": "did:key:agg...",
+    "can": "piece/add",
+    "nb": {
+      "piece": { "/": "commitment...car" }, /* commitment proof for piece */
+      "space": "did:web:web3.storage",      /* space where all segments are joined together */
+      "group": "did:web:free.web3.storage", /* grouping of joining segments into an aggregate */
+    }
+  }],
+  "prf": [],
+  "sig": "..."
+}
+```
+
+Aggregator MUST issue a signed receipt with the result of the task. Arranged aggregate for piece receipt looks like:
+
+```json
+{
+  "ran": "bafy...arrange",
+  "out": {
+    "ok": {
+        "status": "accepted",
+        "piece": { "/": "commitment...car" }, /* commitment proof for piece */
+        "aggregate": { "/": "commitment...aggregate-proof" } /* commitment proof */
+    }
+  },
+  "meta": {},
+  "iss": "did:key:agg...",
+  "prf": []
+}
+```
+
+If offered piece is invalid, reason is also reported:
+
+```json
+{
+  "ran": "bafy...invocation",
+  "out": {
+    "error": {
+      "reason": "reasonCode",
+    },
+  },
+  "meta": {},
+  "iss": "did:key:agg...",
+  "prf": []
+}
+```
+
+### `aggregate/add`
+
+An aggregator principal can invoke a capabilty to add an aggregate that is ready to be included in Filecoin deal(s). See [schema](#aggregateadd-schema).
+
+> `did:web:filecoin.web3.storage` invokes capability from `did:web:spade.storage`
+
+```json
+{
+  "iss": "did:web:filecoin.web3.storage",
+  "aud": "did:web:spade.storage",
+  "att": [{
+    "with": "did:web:filecoin.web3.storage",
+    "can": "aggregate/add",
+    "nb": {
+      "offer": { "/": "bafy...many-cars" }, /* dag-cbor CID with offer content */
```

Comment:
```suggestion
      "pieces": { "/": "bafy...many-cars" }, /* dag-cbor CID with content pieces */
```

I think it would make sense to call it `pieces` instead, or alternatively `content` which seems a lot more clear IMO